### PR TITLE
Add debug, eq, .. traits to public structs

### PR DESCRIPTION
**Pull request by ali-bahjati**

### CODE REVIEW

**Review by Reisen — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ pub enum AccountType
 }
 
 /// The current status of a price feed.
-#[derive(Copy, Clone, PartialEq, BorshSerialize, BorshDeserialize, Debug)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
```
Comment (drozdziak1) [inline src/lib.rs:48]: For extra credit, you might want to consider making the derives conditional

https://stackoverflow.com/questions/42046327/conditionally-derive-based-on-feature-flag

**Review by drozdziak1 — COMMENTED** [review]
Lookin good :)

File: src/lib.rs (hunk 1 above)
Comment (drozdziak1) [reply src/lib.rs:48]: (The derived traits may result in more codegen occuring, but I have no evidence of real impact on build artifact size)

Comment (ali-bahjati) [reply src/lib.rs]: I'll test it to make sure the build artifact size won't change. I don't know how smart actually rust is. I did a simple test and it didn't add anything, will test more.

**Review by jayantk — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (Reisen) [reply src/lib.rs]: LTO is pretty good at removing these but what we could do is `#[cfg_attr(feature = "serde")]`

Comment (ali-bahjati) [reply src/lib.rs]: I checked and binary size didn't change (in case of no usage)

**Pull request merged**